includes/discovery/mempools/moxa-etherdevice.inc.php

```diff
@@ -12,12 +12,20 @@
  */
 
 if ($device['os'] == 'moxa-etherdevice') {
-    echo 'Moxa EtherDevice';
+    d_echo('Moxa EtherDevice');
 
-    $total = snmp_get($device, "totalMemory.0", '-OvQ', 'MOXA-IKS6726A-MIB');
-    $avail = snmp_get($device, "freeMemory.0", '-OvQ', 'MOXA-IKS6726A-MIB');
+    // Moxa people enjoy creating similar MIBs for each model!
+    if ($device['sysDescr'] == 'IKS-6726A-2GTXSFP-T') {
+        $mibmod = 'MOXA-IKS6726A-MIB';
+    } else if ($device['sysDescr'] == 'EDS-G508E-T') {
+        $mibmod = 'MOXA-EDSG508E-MIB';
+    }
+
+    $mem_res = snmp_get_multi_oid($device, 'totalMemory.0 freeMemory.0', '-OQUs', $mibmod);
+    $total = $mem_res['totalMemory.0'];
+    $avail = $mem_res['freeMemory.0'];
 
     if ((is_numeric($total)) && (is_numeric($avail))) {
-        discover_mempool($valid_mempool, $device, 0, 'moxa-etherdevice-mem', 'Memory', '1', null, null);
+        discover_mempool($valid_mempool, $device, 0, 'moxa-etherdevice', 'Memory', '1', null, null);
```

Comment:
The removal of `-mem` will lead to loss in data so you'll need to leave this as it was,

Reply:
Yeah, I didn't want to have a silly file name under polling dir. And that is totally my mistake, I'm sorry.

Reply:
It's cool, It should be how you changed it to but 'legacy code' :(